src/Serialization/Fields.jl

```diff
@@ -226,30 +226,38 @@ end
 
 ################################################################################
 # FracField
-function save_internal(s::SerializerState, K::FracField)
+
+encodeType(::Type{<:AbstractAlgebra.Generic.FracField}) = "AbstractAlgebra.Generic.FracField"
+reverseTypeMap["AbstractAlgebra.Generic.FracField"] = AbstractAlgebra.Generic.FracField
+
+function save_internal(s::SerializerState, K::AbstractAlgebra.Generic.FracField)
     return Dict(
         :base_ring => save_type_dispatch(s, base_ring(K)),
     )
 end
 
 function load_internal(s::DeserializerState,
-                       ::Type{<: FracField},
+                       ::Type{<: AbstractAlgebra.Generic.FracField},
                        dict::Dict)
     R, _ = load_unknown_type(s, dict[:base_ring])
 
     return FractionField(R, cached=false)
 end
 
 # elements
-function save_internal(s::SerializerState, f::FracElem)
+encodeType(T::Type{<:AbstractAlgebra.Generic.FracElem}) =
+  T == fmpq ? "fmpq" : "AbstractAlgebra.Generic.FracElem"
```

Comment:
I feel a bit weird about this line, maybe there is a better way I am not seeing?

Reply:
Yes, just do `encodeType(::fmpq) = "fmpq"`. Or do you want to insert the name? Then it would just be `sprint(show, T)`?